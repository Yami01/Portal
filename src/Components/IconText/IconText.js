/**
 *
 * IconText
 * @format
 *
 */

import type {AbstractComponent, Node} from 'react'
import React, {memo} from 'react'
import {View} from 'react-native'
import {TouchableRipple} from 'react-native-paper'
import IconTextStyles from './styles'
import type {PropsType} from './types'
import {Colors} from '@/Theme/Variables'
import Text from '@/Components/Text'
import Icon from '@/Components/Icon'

const IconText: AbstractComponent<PropsType> = memo(
  (props: PropsType): Node => {
    const { label, labelColor = Colors.white, icon, testID, onPress } = props

    const labelStyle = [IconTextStyles.label, { color: labelColor }]

    return (
      <TouchableRipple
        {...testID}
        style={IconTextStyles.container}
        disabled={!onPress}
        onPress={onPress}
      >
        <View style={IconTextStyles.container}>
          <Text fontWeight={'bold'} style={labelStyle} numberOfLines={1}>
            {label}
          </Text>
          {icon && (
            <Icon
              fillColor={icon.fillColor}
              group={icon.group}
              name={icon.name}
              width={icon.width}
              height={icon.height}
            />
          )}
        </View>
      </TouchableRipple>
    )
  },
)

IconText.displayName = 'IconText'

export default IconText
