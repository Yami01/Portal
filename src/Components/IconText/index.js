/**
 *
 * @flow
 *
 */

export {default} from './IconText';
