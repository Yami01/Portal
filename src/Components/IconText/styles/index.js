/**
 *
 * @format
 *
 */

import {StyleSheet} from 'react-native'

import type {StyleType} from '../types'

const IconTextStyles: StyleType = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-start',
  },
  label: {
    fontSize: 17,
    lineHeight: 24,
  },
})

export default IconTextStyles
