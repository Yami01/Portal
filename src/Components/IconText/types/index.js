/**
 *
 * @format
 *
 */

import type {TextStyleProp, ViewStyleProp,} from 'react-native/Libraries/StyleSheet/StyleSheet'
import type {PressEvent} from 'react-native/Libraries/Types/CoreEventTypes'

export type IconType = {
  group: string,
  name: string,
  width?: number,
  height?: number,
  fillColor?: string,
}

export type PropsType = {
  label?: string,
  labelColor?: string,
  icon?: IconType,
  testID?: string,
  onPress?: (event: PressEvent) => any,
}

export type StyleType = {
  container: ViewStyleProp,
  label: TextStyleProp,
}
