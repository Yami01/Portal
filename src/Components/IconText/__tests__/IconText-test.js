/**
 *
 * IconText Test
 * @format
 * @flow
 *
 */

import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'

import { WHITE } from '@colors'
import IconText from '../IconText'

describe('IconText', () => {
  const label = 'Test Icon Label'
  const labelColor = '#123456'
  const centerIcon = {
    fillColor: WHITE,
    group: 'general',
    name: 'arrow_drop_down',
  }
  const onPressMock = jest.fn()

  test('IconText render', () => {
    const tree = render(
      <IconText label={label} icon={centerIcon} onPress={onPressMock} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('IconText Disabled', () => {
    const { queryByRole } = render(<IconText label={label} />)

    const buttonByRole = queryByRole('button')
    expect(buttonByRole).toBeNull()
  })

  test('IconText render properly with label color', () => {
    const { getByText } = render(
      <IconText label={label} labelColor={labelColor} />,
    )

    const labelNode = getByText(label)
    expect(labelNode).not.toBeNull()
    expect(labelNode).toHaveStyle({ color: labelColor })
  })

  test('IconText Enabled', () => {
    const { getByRole } = render(
      <IconText label={label} icon={centerIcon} onPress={onPressMock} />,
    )

    const button = getByRole('button')
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalled()
    expect(() => {
      fireEvent.press(button)
    }).not.toThrow()
  })
})
