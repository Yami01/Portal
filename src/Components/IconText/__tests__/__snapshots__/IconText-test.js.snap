// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconText IconText render 1`] = `
<View
  accessibilityHint="Center Header Hint"
  accessibilityLabel="Center Header"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          undefined,
          Array [
            Object {
              "fontSize": 17,
              "lineHeight": 24,
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }
    >
      Test Header Label
    </Text>
    <View
      accessibilityHint="This button will take to the notification"
      accessibilityLabel="Accessible  header button 1"
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <svg
        data-file-name="SvgArrowDropDown"
        fill="#fff"
      />
    </View>
  </View>
</View>
`;
